projectDocs/dev/developerGuide/developerGuide.md

```diff
@@ -1445,3 +1445,238 @@ Please see the `EventExtensionPoints` class documentation for more information,
+
+With modal dialogs, the easiest way to respond to user input is via the return code.
+
+```py
+saveDialog = MessageDialog(
+	mainFrame,
+	"Would you like to save your changes before exiting?",
+	"Save changes?",
+	buttons=DefaultButtonSet.SAVE_NO_CANCEL
+)
+
+match saveDialog.ShowModal():
+	case ReturnCode.SAVE:
+		...  # Save the changes and close
+	case ReturnCode.NO:
+		...  # Discard changes and close
+	case ReturnCode.CANCEL:
+		...  # Do not close
+```
+
+For non-modal dialogs, the easiest way to respond to the user pressing a button is via callback methods.
+
+```py
+def read_changelog():
+	...  # Do something
+
+def download_update():
+	...  # Do something
+
+def remind_later():
+	...  # Do something
+
+updateDialog = MessageDialog(
+	mainFrame,
+	"An update is available. "
+	"Would you like to download it now?",
+	"Update",
+	buttons=None,
+).addButton(
+	ReturnCode.YES,
+	label="Yes",
+	default_focus=True,
+	callback=download_update
+).addButton(
+	ReturnCode.NO,
+	default_action=True,
+	label="Remind me later",
+	callback=remind_later
+).addButton(
+	ReturnCode.HELP,
+	closes_dialog=False,
+	label="What's new",
+	callback=read_changelog
+)
+
+updateDialog.Show()
+```
+
+You can set many of the parameters to `addButton` later, too:
+
+* The default focus can be set by calling `setDefaultFocus` on your message dialog instance, and passing it the ID of the button to make the default focus.
+* The fallback action can be set later by calling `setFallbackAction` or `SetEscapeId` with the ID of the button which performs the fallback action.
+* The button's label can be changed by calling `setButtonLabel` with the ID of the button and the new label.
+
+#### Fallback actions
+
+It is worth interrogating the fallback action further.
+The fallback action is the action performed when the dialog is closed without the user pressing one of the buttons you added to the dialog.
+This can happen for several reasons:
+
+* The user pressed `esc` or `alt+f4` to close the dialog.
+* The user used the title bar close button or system menu close item to close the dialog.
+* The user closed the dialog from the Task View, Taskbar or App Switcher.
+* The user is quitting NVDA.
+* Some other part of NVDA or an add-on has asked the dialog to close.
+
+By default, the fallback action is set to `EscapeCode.CANCEL_OR_AFFIRMATIVE`.
+This means that the fallback action will be the cancel button if there is one, the ok button if there is no cancel button but there is an ok button[^fn_defaultFallbackOk], or otherwise `None`.
+The fallback action can also be set to `EscapeCode.NO_FALLBACK` to disable closing the dialog like this entirely.
+If it is set to any other value, the value must be the id of a button to use as the default action.
+
+[^fn_defaultFallbackOk]: In actuality, the value of `dialog.GetAffirmativeId()` is used to find the button to use as fallback if using `EscapeCode.CANCEL_OR_AFFIRMATIVE` and there is no button with `id=ReturnCode.CANCEL`.
```

Comment:
these references (e.g. `[^fn_defaultFallbackOk]`) don't appear to work in the built HTML from appveyor